From 3b625824089c27a64308feb41d7cdcaf37af79fd Mon Sep 17 00:00:00 2001
From: bodapati <shalini.salomi.bodapati@intel.com>
Date: Tue, 5 May 2020 21:00:15 +0530
Subject: [PATCH] Enabled AVX2 and FMA compiler Flags for libart module.

Performance Impact:
10-12% Launch Time performance improvement for top ranked apps
13-15% Cold Launch Time performance improvement for Top Ranked Education Apps
11-14% Warm Launch Time performance improvement for Top Ranked Education Apps

Test: run-test gtest

Tracked-On: https://jira.devtools.intel.com/browse/OAM-90793
Change-Id: I36ba4a19082980c859044617a5ef5a76ba8936e0
Signed-off-by: bodapati <shalini.salomi.bodapati@intel.com>
---
 build/Android.bp | 21 ++++++++++++++++++++-
 1 file changed, 20 insertions(+), 1 deletion(-)

diff --git a/build/Android.bp b/build/Android.bp
index 7b807d5..aecfadd 100644
--- a/build/Android.bp
+++ b/build/Android.bp
@@ -23,7 +23,7 @@ art_clang_tidy_errors = [
     "android-cloexec-open",
     "bugprone-argument-comment",
     "bugprone-lambda-function-name",
-    "bugprone-unused-raii",  // Protect scoped things like MutexLock.
+    "bugprone-unused-raii", // Protect scoped things like MutexLock.
     "bugprone-unused-return-value",
     "bugprone-virtual-near-miss",
     "modernize-use-bool-literals",
@@ -108,6 +108,25 @@ art_global_defaults {
         "-D_LIBCPP_ENABLE_THREAD_SAFETY_ANNOTATIONS",
     ],
 
+    arch: {
+        x86: {
+            avx2: {
+                cflags: [
+                    "-mavx2",
+                    "-mfma",
+                ],
+            },
+        },
+        x86_64: {
+            avx2: {
+                cflags: [
+                    "-mavx2",
+                    "-mfma",
+                ],
+            },
+        },
+    },
+
     target: {
         android: {
             cflags: [
-- 
2.7.4

